tests/transforms/test_experimental/test_transform_program.py

```diff
@@ -195,6 +252,226 @@ def test_valid_transforms(self):
+        assert postprocessing is null_postprocessing
+
+        obj = [1, 2, 3, "b"]
+        assert null_postprocessing(obj) is obj
+
+    def test_informative_transforms_not_supported(self):
+        """Test that a program with an informative raises a `NotImplementedError` on call."""
+        my_transform = TransformContainer(first_valid_transform, is_informative=True)
+        prog = TransformProgram((my_transform,))
+        batch = (qml.tape.QuantumScript([], [qml.state()]),)
+
+        with pytest.raises(
+            NotImplementedError, match="Informative transforms are not yet supported."
+        ):
+            prog(batch)
+
+    def test_cotransform_support_notimplemented(self):
+        """Test that a transform with a cotransfrom raises a not implemented error."""
+
+        my_transform = TransformContainer(
+            first_valid_transform, classical_cotransform=lambda res: res
+        )
+
+        prog = TransformProgram((my_transform,))
+
+        batch = (qml.tape.QuantumScript([], [qml.state()]),)
+        with pytest.raises(
+            NotImplementedError, match="cotransforms are not yet integrated with TransformProgram"
+        ):
+            prog(batch)
+
+    def test_single_transform_program(self):
+        """Basic test with a single transform that only modifies the tape but not the results."""
+
+        def single_null_postprocessing(results):
+            return results[0]
+
+        def remove_operation_at_index(
+            tape: qml.tape.QuantumTape, index: int
+        ) -> (Sequence[qml.tape.QuantumTape], Callable):
+            """A valid transform."""
+            new_ops = list(tape.operations)
+            new_ops.pop(index)  # pylint:disable=protected-access
+            return (
+                qml.tape.QuantumScript(new_ops, tape.measurements, shots=tape.shots),
+            ), single_null_postprocessing
+
+        container = TransformContainer(remove_operation_at_index, kwargs={"index": 1})
+        prog = TransformProgram((container,))
+
+        tape0 = qml.tape.QuantumScript(
+            [qml.S(0), qml.T(1), qml.SX(2)], [qml.expval(qml.PauliZ(0))], shots=100
+        )
+        batch = (tape0,)
+        new_batch, fn = prog(batch)
+
+        assert len(new_batch) == 1
+        expected = [qml.S(0), qml.SX(2), qml.expval(qml.PauliZ(0))]
+        for op1, op2 in zip(expected, new_batch[0]):
+            assert qml.equal(op1, op2)
+        assert new_batch[0].shots == qml.measurements.Shots(100)
+
+        assert fn.func is _apply_postprocessing_stack
+        assert fn.args == tuple()
+
+        assert len(fn.keywords["postprocessing_stack"]) == 1
+        postprocessing0 = fn.keywords["postprocessing_stack"][0]
+        assert postprocessing0.func is _batch_postprocessing
+        assert postprocessing0.args == tuple()
+        assert postprocessing0.keywords["individual_fns"] == [single_null_postprocessing]
+        assert postprocessing0.keywords["batch_slices"] == [slice(0, 1)]
```

Comment:
I'm learning so much about checking function signatures.

Reply:
me too. It was always a pet peeve of mine to not know what the locally defined postprocessing functions depended on.  I'm thinking using `partial` and a global function will allow us to be much more explicit about what the post-processing is doing.

Reply:
to be clear, this only works with `partial` funcs. Otherwise you'll need to use the `inspect` module

Reply:
For functions that rely on closure, we could also check `obj.__closure__`, but I think I like using a partial a little bit better than a local function with closure.